### DISCUSSION TITLE
Site to Site Ipsec tunnel both ends behind NAT #1191

### DISCUSSION THREAD

**Discussion by aspx1981** [strongswan/strongswan#1223] [General]
Hello All,

I'm writing here because I spent 3 weeks without any luck to make an IPSec tunnel working.
I got 2 ubuntu servers behind a ISP router each.
I configured client strongswan IPSec vpn on both servers and they are working fine.
But not the site to site config.
here the schema

clientsA <-> wlan0 <serverA> eth0 <-> routerA-eth0 <-> routerA-wan0 <->internet <-> routerB-wan0 <-> routerB-eth0 <-> eth0 <serverB> wlan0 <-> clientsB
192.168.130.0/24 <-> 192.168.130.221 <-> 192.168.0.127 <-> 192.168.0.10 <-> 81........... <-> 81......... <-> 192.168.1.1 <-> 192.168.1.102 <-> 192.168.139.221 <-> 192.168.139.0/24

Both routers are configured to allow and forward port 500 and 4500 and in fact VPN clients work.

here my config:
strongswan.conf - strongSwan configuration ServerA=ServerB

charon {
interfaces_ignore = wlan0
interfaces_use = eth0
load_modular = yes
plugins {
include strongswan.d/charon/*.conf
}
}
include strongswan.d/*.conf
ipsec.conf - strongSwan IPsec configuration file ServerA
basic configuration

config setup
charondebug="ike 2, knl 2, cfg 2, net 2"
uniqueids=yes
conn ikev2-vpn
auto=add
compress=no
type=tunnel
keyexchange=ikev2
fragmentation=yes
forceencaps=yes
dpdaction=clear
dpddelay=300s
rekey=no
left=%any
leftid=@vpn.......freeddns.org
leftcert=vpn-server-cert.pem
leftsendcert=always
leftsubnet=0.0.0.0/0
right=%any
rightid=%any
rightauth=eap-mschapv2
rightsourceip=192.10.10.0/24
rightdns=192.168.0.127
rightsendcert=never
eap_identity=%identity
ike=aes256-sha2_256-modp2048
esp=aes256-sha2_256
reauth=no
aggressive=no
leftfirewall=yes
leftauth=pubkey
conn A-to-B
type=tunnel
ikelifetime=36000s
keylife=8h
rekeymargin=3m
keyingtries=5
mobike=no
lifetime=8h
left=192.168.0.127
leftid=@vpn.........freeddns.org
leftsourceip=192.168.130.221
leftsubnet=192.168.130.0/24
right=vpn...2.freeddns.org
rightid=@vpn.......2.freeddns.org
rightsubnet=192.168.139.0/24
rightsourceip=192.168.139.221
keyexchange=ikev2
authby=secret
ike=aes256-sha2_256-modp2048
esp=aes256-sha2_256
dpddelay=30
dpdtimeout=120
dpdaction=restart
auto=start
rightfirewall=yes
leftfirewall=yes
ipsec.conf - strongSwan IPsec configuration file ServerB
basic configuration

config setup
charondebug="ike 2, knl 2, cfg 2, net 2"
uniqueids=yes
conn ikev2-vpn
auto=add
compress=no
type=tunnel
keyexchange=ikev2
fragmentation=yes
forceencaps=yes
dpdaction=clear
dpddelay=300s
rekey=no
left=%any
leftid=@vpn......2.freeddns.org
leftcert=vpn-server-cert.pem
leftsendcert=always
leftsubnet=0.0.0.0/0
right=%any
rightid=%any
rightauth=eap-mschapv2
rightsourceip=192.10.20.0/24
rightdns=192.168.1.102
rightsendcert=never
eap_identity=%identity
ike=aes256-sha2_256-modp2048
esp=aes256-sha2_256
reauth=no
aggressive=no
leftfirewall=yes
leftauth=pubkey
conn B-to-A
type=tunnel
ikelifetime=36000s
keylife=8h
rekeymargin=3m
keyingtries=5
mobike=no
lifetime=8h
left=192.168.1.102
leftid=@vpn........2.freeddns.org
leftsourceip=192.168.139.221
leftsubnet=192.168.139.0/24
right=vpn....freeddns.org
rightid=@vpn........freeddns.org
rightsubnet=192.168.130.0/24
rightsourceip=192.168.130.221
keyexchange=ikev2
authby=secret
ike=aes256-sha2_256-modp2048
esp=aes256-sha2_256
dpddelay=30
dpdtimeout=120
dpdaction=restart
auto=start
rightfirewall=yes
leftfirewall=yes
--------------------------------
sysconfig
--------------------------------
net.ipv4.ip_forward=1
net.ipv4.conf.all.accept_redirects=0
net.ipv4.conf.all.send_redirects=0
net.ipv4.ip_no_pmtu_disc=1
net.ipv4.conf.eth0.rp_filter=0
net.ipv4.conf.tun0.rp_filter=0
net.ipv4.conf.all.rp_filter=0
net.ipv4.conf.default.rp_filter=0
net.ipv4.conf.lo.rp_filter=0
net.ipv4.conf.all.forwarding=1
net.ipv4.conf.default.forwarding=1
net.ipv4.conf.eth0.forwarding=1
net.ipv4.conf.lo.forwarding=1
net.ipv6.conf.all.forwarding=1
net.ipv6.conf.default.forwarding=1
net.ipv6.conf.eth0.forwarding=1
net.ipv6.conf.lo.forwarding=1
net.ipv4.tcp_fwmark_accept=1

root@...:/etc/strongswan.d# ipsec status
Security Associations (1 up, 0 connecting):
A-to-B[1]: ESTABLISHED 3 seconds ago, 192.168.0.127[vpn.....freeddns.org]...81..........[vpn.....2.freeddns.org]
A-to-B{1}: INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: ca4a0941_i c85c8974_o
A-to-B{1}: 192.168.130.0/24 === 192.168.139.0/24
root@...:/etc/strongswan.d# ip route show table 220
192.168.139.0/24 via 192.168.0.10 dev eth0 proto static src 192.168.130.221

everything seems ok, but no traffic, no ping, nothing.

Thanks,
Bye

**Comment by Thermi** [comment]
please provide all as shown in #196
